Add Get-Location Pester tests

h93xV2 opened the pull request:
# PR Summary

Addresses the issue first raised in #2413 by adding tests to
increase coverage of the Get-Location cmdlet which is
supported by the file system provider.

## PR Context

#2413 was originally opened on October 3, 2016. Since then, the test coverage of Get-Location has remained low.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): <!-- Experimental feature name(s) here -->
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

h93xV2 committed:
Add Get-Location Pester tests

Addresses the issue first raised in #2413 by adding tests to
increase coverage of the Get-Location cmdlet which is
supported by the file system provider.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/PowerShell/PowerShell?pullRequest=11463) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/PowerShell/PowerShell?pullRequest=11463) before we can accept your contribution.<br/><br/>:x: h93xV2 [sign now](https://cla.opensource.microsoft.com/PowerShell/PowerShell?pullRequest=11463)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/PowerShell/PowerShell?pullRequest=11463) it.</sub>

h93xV2 committed:
Adjust Get-Location test to account for OS differences

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
+    }
+
+    It "Should throw an exception when missing an argument for parameter 'PSProvider'" {
+        { Get-Location -PSProvider } | Should -Throw "Missing an argument for parameter 'PSProvider'."
```
We can not use localized strings. Please use `Should -Throw -ErrorId "..."` pattern.
Below too.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
+    It "Should return a PathInfoStack with the correct values for parameter 'Stack'" {
+        $stackAsArray = (Get-Location -Stack).ToArray()
+
+        $stackAsArray.Length -gt 0 | Should -BeTrue
```
```suggestion
        $stackAsArray.Length | Should -BeGreaterThan 0
```

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
+    It "Should return a PathInfoStack with the correct values for the argument for parameter 'StackName'" {
+        $tempDirectory = Join-Path -Path $currentDirectory -ChildPath "getLocationTempDir"
+
+        New-Item -Path ($tempDirectory) -ItemType "directory"
```
```suggestion
        New-Item -Path ($tempDirectory) -ItemType "directory" > $null
```

h93xV2 committed:
Update Get-Location tests peer review feedback

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
 
     AfterEach {
-	Pop-location
+	    Pop-location
```
Indentation.
We use 4 spaces (not tabs) per indentation step. Please fix below too.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
     It "Should do exactly the same thing as its alias" {
-	(pwd).Path | Should -BeExactly (Get-Location).Path
+	    (pwd).Path | Should -BeExactly (Get-Location).Path
+    }
```
Please remove the test - we have another place to tests aliases.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
+
+        Get-Location -PSDrive $tempPSDriveName | Should -BeOfType System.Management.Automation.PathInfo
+
+        $tempPSDrive | Remove-PSDrive
```
If we should cleanup it is better to use try-filnally and put `$tempPSDrive | Remove-PSDrive` to finally block.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
+    }
+
+    It "Should return a PathInfoStack with the correct values for the argument for parameter 'StackName'" {
+        $tempDirectory = Join-Path -Path $currentDirectory -ChildPath "getLocationTempDir"
```
Please use $TestDrive instead of $currentDirectory.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,80 @@
+
+        $stackAsArray[0].Path | Should -BeExactly $tempDirectory
+
+        Remove-Item -Path $tempDirectory
```
With using $TestDrive Pester will automatically remove the temporary directory after each Describe block

h93xV2 committed:
Update Get-Location tests per review

Correct the number of spaces used for indentation from 4 to the standard 2.
Remove alias test because aliases are already tested elsewhere.
Improve test cleanup.
Utilize Pester's $TempDrive for automatic directory cleanup.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -1,21 +1,77 @@
-    BeforeEach {
-	Push-Location $currentDirectory
-    }
+  $currentDirectory=[System.IO.Directory]::GetCurrentDirectory()
```
Please use 4 spaces. It is our formatting convention.

h93xV2 committed:
Update Get-Location Pester tests per review

Correct spacing issue. Spacing was erroneously changed from
4 to 2. 4 spaces is the formatting convention.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1:
```diff
@@ -2,20 +2,76 @@
 
     AfterEach {
-	Pop-location
+        Pop-location
```
Please put line 4 to BeforeAll block.

It seems BeforeEach and AfterEach is used only in one test and we could move the code to the test.

h93xV2 committed:
Update Get-Location Pester tests per review

Convert BeforeEach and AfterEach to BeforeAll and AfterAll.

h93xV2 committed:
Update Get-Location Pester tests per review

Refactor test setup.

h93xV2 committed:
Update Get-Location Pester tests

Fix Windows CI build fail reason.

h93xV2 commented:
@adityapatwardhan Can this pull request be merged? I'm not authorized to do so . . .

iSazonov commented:
@h93xV2 Could you please reopen? The PR was approved and waiting to merge.
